ci/gitlab/.gitlab-ci.yml

```diff
@@ -85,6 +85,49 @@ nop11:
   script:
     - ./scripts/test.sh
 
+scan-build:
+  stage: test
+  only:
+    - schedules
+  # prepare build to be submitted for static code analysis
+  variables:
+    TEST_BUILD_DIR: 'scan-build'
+    TEST_CMAKE_BUILD_TYPE: 'Debug'
+    TEST_INSTALL_DESTDIR: "$CI_PROJECT_DIR/scan-build/bin"
+    TEST_WITH_TESTSUITE: '0'
+    TEST_WITH_SOTA_TOOLS: '0'
+    TEST_WITH_OSTREE: '0'
+    TEST_WITH_DEB: '0'
+    TEST_WITH_ISOTP: '0'
+    TEST_WITH_PARTIAL: '0'
+  image: "$UBUNTU_BIONIC_PR_IMAGE"
+  script:
+    - mkdir -p $TEST_INSTALL_DESTDIR
+    - ./scripts/test.sh
+    - git archive -o scan.tar HEAD
+  artifacts:
+    paths:
+      - scan-build/bin
+      - scan.tar
+
+veracode-scan:
+  stage: deploy
+  only:
+    - schedules
+  dependencies:
+    - scan-build
+  image: "$UBUNTU_BIONIC_PR_IMAGE"
+  before_script:
+    - apt update
```

Comment:
It looks like you actually do not need our Docker image here, you could pull an image with the correct version of java preinstalled and not worry about installing it with apt in the pipeline.

Reply:
Which image is that? Got link/name?

Reply:
https://hub.docker.com/_/openjdk ?

Reply:
AFAIK, gitlab can serve as a docker image registry, we could prepare our own images, put them into gitlab so CI pipelines/test can utilize them. Effectively, each aktualizr docker file 'bakes' dynamically/on the fly during test running, while having pre-baked docker image could speed these tests little bit.